rules/jsll/tests/jsll-validate-config/test.ts

```diff
@@ -0,0 +1,155 @@
+import { generateHTMLPage } from 'sonarwhal/dist/tests/helpers/misc';
+import { getRuleName } from 'sonarwhal/dist/src/lib/utils/rule-helpers';
+import { IRuleTest } from 'sonarwhal/dist/tests/helpers/rule-test-type';
+import * as ruleRunner from 'sonarwhal/dist/tests/helpers/rule-runner';
+
+import { code, deleteProp, modifyConfigVal, scriptWrapper } from '../helpers/common';
+
+const ruleName = getRuleName(__dirname);
+const messages = {
+    invalidAppId: `The "appId" must be a non-empty string.`,
+    invalidAutoCapture: `The "autoCapture" property is not a valid object.`,
+    invalidCountryCode: `The "market" parameter contains invalid country code "oo".`,
+    invalidLangCode: `The "market" parameter contains invalid language code "oo".`,
+    invalidLineage: `"lineage" parameter is not set to true.`,
+    invalidMarketValue: `The format of "market" parameter is not valid.`,
+    invalidUseShortNameForContentBlob: `"useShortNameForContentBlob" parameter is not set to true.`,
+    missingAppId: `appId is required but missing.`,
+    missingAutoCapture: `autoCapture is required but missing.`,
+    missingLineageAndScroll: `scroll, lineage are optional but missing.`,
+    missingOptionalConfigProp: `useShortNameForContentBlob is optional but missing.`,
+    missingPageName: `pageName is optional but missing.`,
+    missingPageNameAndEnv: `pageName, env are optional but missing.`,
+    missingRequiredConfigProps: `autoCapture, coreData are required but missing.`,
+    missingUseShortNameForContentBlob: `useShortNameForContentBlob is optional but missing.`,
+    noConfigArgs: `No config argument was passed to JSLL initialization function "awa.init".`
+};
+const stringPropertyConfig = {
+    'autoCapture': { // eslint-disable-line quote-props
```

Comment:
why does the property have quotes?

Reply:
@sarvaje For test purposes, espree parse the properties with or without quotes as different types. Although we don't use the parsed js to validate `config` any more (now we inject js script to validate `config` on scan end, this way we can better handle the "calculated" config values, see https://github.com/Microsoft/sonarwhal-rules-microsoft/pull/6#issuecomment-360610615), I still kept those tests there because why not 😄 .

Reply:
ok! :)